### PR TITLE
DFPL-760: Allow Child Solicitor and Barrister to upload docs

### PR DESCRIPTION
**Pull request by AlistairEdwinOsborne**
https://tools.hmcts.net/jira/browse/DFPL-760

Allow child solicitor, barrister and labarrister to be able to upload documents in the manage documents flow

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsController.java [hunk 1]
```diff
@@ -298,6 +300,10 @@ private DocumentUploaderType getUploaderType(Long id) {
             return DocumentUploaderType.SOLICITOR;
         }
 
+        if (caseRoles.stream().anyMatch(barristers()::contains)) {
+            return DocumentUploaderType.BARRISTER;
+        }
+
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsController.java:306]: Is it possible to move them before line 285? And the change of line 286 will not be needed.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsControllerSubmittedTest.java [hunk 2]
```diff
@@ -81,6 +81,18 @@ void shouldSendEmailsWhenAnyOtherDocumentUploadedBySolicitor() throws Notificati
             TEST_CASE_ID);
     }
 
+    // Same behaviour as solicitior for any document but with barrister user
+    @Test
+    void shouldSendEmailsWhenAnyOtherDocumentUploadedByLaBarrister() throws NotificationClientException {
+        when(featureToggleService.isNewDocumentUploadNotificationEnabled()).thenReturn(true);
+        when(idamClient.getUserDetails(any())).thenReturn(UserDetails.builder().build());
+        givenCaseRoles(TEST_CASE_ID, USER_ID, CaseRole.LABARRISTER);
```
Comment (so99y) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsControllerSubmittedTest.java:89]: Test case for CaseRole.BARRISTER?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsControllerSubmittedTest.java [hunk 3]
```diff
@@ -93,6 +105,18 @@ void shouldSendEmailsWhenAnyOtherDocumentFromHearingsUploadedBySolicitor()
             TEST_CASE_ID);
     }
 
+    // Check above test with barrister as behaviour will be identical
+    @Test
+    void shouldSendEmailsWhenAnyOtherDocumentFromHearingsUploadedByLaBarrister()
+        throws NotificationClientException {
+        when(featureToggleService.isNewDocumentUploadNotificationEnabled()).thenReturn(true);
+        given(idamClient.getUserDetails(any())).willReturn(UserDetails.builder().build());
+        givenCaseRoles(TEST_CASE_ID, USER_ID, CaseRole.LABARRISTER);
```
Comment (so99y) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsControllerSubmittedTest.java:114]: Test case for CaseRole.BARRISTER?

**Review by so99y — CHANGES_REQUESTED** [review]
please see the comments.

**Review by SomeshRA — APPROVED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsController.java [hunk 4]
```diff
@@ -282,6 +283,7 @@ public void handleSubmitted(@RequestBody CallbackRequest request) {
         DocumentUploaderType userType = getUploaderType(caseData.getId());
 
         if (!DocumentUploaderType.SOLICITOR.equals(userType)
+            && !DocumentUploaderType.BARRISTER.equals(userType)
             || this.featureToggleService.isNewDocumentUploadNotificationEnabled()) {
```
Comment (so99y) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsController.java:287]: Maybe 
```
this.featureToogleService.isNewDocumentUploadNotfiicationEnabled() || 
(!DocumentUploaderType.SOLICITOR.equals(userType)
            && !DocumentUploaderType.BARRISTER.equals(userType))
```

**Review by so99y — DISMISSED** [review]
All good

**Review by so99y — APPROVED** [review]

**Pull request merged**